bin/stencil-start.js

```diff
@@ -87,45 +87,66 @@ if (!(dotStencilFile.normalStoreUrl) || !(dotStencilFile.customLayouts)) {
     process.exit(2);
 }
 
-let staplerUrl;
-const headers = {
-    'stencil-cli': Pkg.version,
-};
-if (dotStencilFile.staplerUrl) {
-    staplerUrl = dotStencilFile.staplerUrl;
-    headers['stencil-store-url'] = dotStencilFile.normalStoreUrl;
-} else {
-    staplerUrl = dotStencilFile.normalStoreUrl;
-}
+runAPICheck(dotStencilFile, Pkg.version)
+    .then(storeInfoFromAPI => {
+        dotStencilFile.storeUrl = storeInfoFromAPI.sslUrl;
+        dotStencilFile.normalStoreUrl = storeInfoFromAPI.baseUrl;
+        dotStencilFile.stencilServerPort = stencilServerPort;
+    })
+    .then(() => {
+        startServer();
+    })
+    .catch(err => console.error(err.message));
 
-Wreck.get(
-    Url.resolve(staplerUrl, '/stencil-version-check?v=' + Pkg.version),
-    {
-        headers: headers,
-        json: true,
-        rejectUnauthorized: false,
-    },
-    function (err, res, payload) {
-        if (err || !payload) {
-            console.error(
-                'The BigCommerce Store you are pointing to either does not exist or is not available at this time.'.red,
-            );
-        } else if (payload.error) {
-            return console.error(payload.error.red);
-        } else if (payload.status !== 'ok') {
-            console.error(
-                'Error: You are using an outdated version of stencil-cli, please run '.red +
-                '$ npm install -g @bigcommerce/stencil-cli'.cyan,
-            );
-        } else {
-            dotStencilFile.storeUrl = payload.sslUrl;
-            dotStencilFile.normalStoreUrl = payload.baseUrl;
-            dotStencilFile.stencilServerPort = stencilServerPort;
+/**
+ *
+ * @param {object} stencilConfig
+ * @param {string} currentCliVersion
+ * @returns {Promise<object>}
+ */
+async function runAPICheck (stencilConfig, currentCliVersion) {
+    const staplerUrl = stencilConfig.staplerUrl
+        ? stencilConfig.staplerUrl
+        : stencilConfig.normalStoreUrl;
+    const reqUrl = Url.resolve(staplerUrl, '/stencil-version-check?v=' + currentCliVersion);
+    let response, payload;
```

Comment:
But payload is used out of try-catch, so if I defile it as const inside try-catch - it will be unavailable outside try-catch.
Though respose can be defined as const, I didn’t notice that it isn’t used outside of try-catch